### ISSUE TITLE
[C++ Frontend] Question about tensor's item() method

### ISSUE THREAD

**Issue by blackccpie** [pytorch/examples#498]
Hi,

I've a question regarding the use of `torch::Tensor::item()` method in the mnist sample code.
If I look at the python doc about `item`, it says it applies to single value tensors, but in the mnist sample it is applied to loss tensor (l.80 of `mnist.cpp`), but are loss tensors always single value tensors?

And by the way, how come, concerning the loss function, default mean reduction is used for training and sum reduction is used for testing?

thx for your help,

Regards

Albert

**Comment by soumith** [comment]
yes, loss Tensors are always scalars (i.e.e single value Tensors) by default. We use mean reduction by default.
